### DISCUSSION TITLE
Is it possible to render request examples for webhooks?

### DISCUSSION THREAD

**Discussion by AnnaKudriasheva** [scalar/scalar#3122] [Q&A]
We want to show request samples for webhooks to our users that you'd typically see on the right side of the screen. Is there a way to do it right now? That's something that Redocly currently supports:

OpenAPI 3.1 webhook example:
```
webhooks:
  userCreated:
    post:
      summary: Notify when a user is created
      requestBody:
        description: Data about the newly created user
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/User'
            examples:
              id: "123
              name: "Jane Doe"
              email: "jane.doe@example.com"
              age: 29
      responses:
        '200':
          description: Webhook successfully processed
        '400':
          description: Invalid data
``` 

Thank you!

**Comment by hanspagel** [comment]
Currently not, but let’s add this! New issue: https://github.com/scalar/scalar/issues/3203

**Comment by Catzilla** [comment]
It would be great to render request parameters as well (like Headers and Query).

According to [OpenAPI 3.1 Specification](https://spec.openapis.org/oas/v3.1.0#fixed-fields), webhook item is a [Path Item Object](https://spec.openapis.org/oas/v3.1.0#path-item-object), identical to used in `paths` property, so I think other properties of this object should be supported as well.

In my particular case, which brought me to this discussion, we add header with signature, that may be used to verify the webhook indeed comes from us:

```yaml
webhooks:
  WebhookEvent:
    post:
      summary: Webhook event
      parameters:
        - name: Webhook-Signature
          in: header
          required: true
          description: Webhook signature
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/WebhookEvent'
```

**Reply by antlio** [reply]
thanks for sharing @Catzilla! adding this in our discussion regarding the topic!
